src/BlockTemplatesController.php

```diff
@@ -335,16 +336,12 @@ function( $template ) {
 				if ( ! $template->description ) {
 					$template->description = BlockTemplateUtils::get_block_template_description( $template->slug );
 				}
-
 				if ( str_contains( $template->slug, 'single-product' ) ) {
 					if ( ! is_admin() && ! BlockTemplateUtils::template_has_legacy_template_block( $template ) ) {
-
 						$new_content       = SingleProductTemplateCompatibility::add_compatibility_layer( $template->content );
 						$template->content = $new_content;
```

Comment:
Since we're tidying this up:

```suggestion
						$template->content = SingleProductTemplateCompatibility::add_compatibility_layer( $template->content );
```